reframe/core/pipeline.py

```diff
@@ -949,19 +950,20 @@ def info(self):
 
         return ret
 
-    def supports_system(self, partition_name):
-        if '*' in self.valid_systems:
-            return True
-
-        if self.current_system.name in self.valid_systems:
-            return True
-
+    def supports_system(self, name):
         # Check if this is a relative name
-        if partition_name.find(':') == -1:
-            partition_name = '%s:%s' % (self.current_system.name,
-                                        partition_name)
+        if name.find(':') != -1:
+            system_name, partition_name = name.split(':')
+        else:
+            system_name, partition_name = self.current_system.name, name
+
+        names_to_test = ['*', '*:*']
+        names_to_test.append(system_name)
+        names_to_test.append('%s:*' % system_name)
+        names_to_test.append('*:%s' % partition_name)
+        names_to_test.append('%s:%s' % (system_name, partition_name))
```

Comment:
Just create the list all at once and use f-strings. Also better call it `valid_matches`:

```suggestion
        valid_matches = ['*', '*:*', f'{system_name}:*', ...]
```

It could also be a set since depending on the input `name`, you may have duplicates.